### PR TITLE
Modify the memory usage rules of the mo hash map

### PR DESCRIPTION
**Pull request by m-schen**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

## Which issue(s) this PR fixes:

issue #5946 #6178

## What this PR does / why we need it:

### CODE REVIEW

File: pkg/container/hashtable/int64_hash_map.go [hunk 1]
```diff
@@ -142,76 +146,119 @@ func (ht *Int64HashMap) FindBatchWithRing(n int, zValues []int64, hashes []uint6
 }
 
 func (ht *Int64HashMap) findCell(hash uint64, key uint64) *Int64HashMapCell {
-	for idx := hash & ht.cellCntMask; true; idx = (idx + 1) & ht.cellCntMask {
-		cell := &ht.cells[idx]
-		if cell.Key == key || cell.Mapped == 0 {
-			return cell
+	if len(ht.rawData) == 1 {
```
Comment (fengttt) [inline pkg/container/hashtable/int64_hash_map.go:149]: Do you really need this if switch?

**Review by fengttt — COMMENTED** [review]
I think the if/else check are not necessary.

File: pkg/container/hashtable/int64_hash_map.go (hunk 1 above)
Comment (m-schen) [reply pkg/container/hashtable/int64_hash_map.go:149]: you're right. it is not necessary here.

**Comment by m-schen** [comment]
> I think the if/else check are not necessary.

resize is necessary to check if one block or not. others are not necessary.

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**